pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -0,0 +1,257 @@
+            copy_var_list_values(fix_nlp.MindtPy_utils.variable_list,
+                                 solve_data.mip.MindtPy_utils.variable_list,
+                                 config)
+            self.add_lazy_oa_cuts(
+                solve_data.mip, dual_values, solve_data, config, opt)
+
+    def handle_lazy_NLP_subproblem_infeasible(self, fix_nlp, solve_data, config, opt):
+        """Solve feasibility problem, add cut according to strategy.
+
+        The solution of the feasibility problem is copied to the working model.
+        """
+        # TODO try something else? Reinitialize with different initial
+        # value?
+        config.logger.info('NLP subproblem was locally infeasible.')
+        for c in fix_nlp.component_data_objects(ctype=Constraint):
+            rhs = ((0 if c.upper is None else c.upper)
+                   + (0 if c.lower is None else c.lower))
+            sign_adjust = 1 if value(c.upper) is None else -1
+            fix_nlp.dual[c] = (sign_adjust
+                               * max(0, sign_adjust * (rhs - value(c.body))))
+        dual_values = list(fix_nlp.dual[c]
+                           for c in fix_nlp.MindtPy_utils.constraint_list)
+
+        if config.strategy == 'OA':
+            config.logger.info('Solving feasibility problem')
+            if config.initial_feas:
+                # config.initial_feas = False
+                feas_NLP, feas_NLP_results = solve_NLP_feas(solve_data, config)
+                # In OA algorithm, OA cuts are generated based on the solution of the subproblem
+                # We need to first copy the value of variables from the subproblem and then add cuts
+                copy_var_list_values(feas_NLP.MindtPy_utils.variable_list,
+                                     solve_data.mip.MindtPy_utils.variable_list,
+                                     config)
+                self.add_lazy_oa_cuts(
+                    solve_data.mip, dual_values, solve_data, config, opt)
+
+    def __call__(self):
+        solve_data = self.solve_data
+        config = self.config
+        opt = self.opt
+        master_mip = self.master_mip
+        cpx = opt._solver_model  # Cplex model
+
+        self.handle_lazy_master_mip_feasible_sol(
+            master_mip, solve_data, config, opt)
+
+        # solve subproblem
+        # Solve NLP subproblem
+        # The constraint linearization happens in the handlers
+        fix_nlp, fix_nlp_result = solve_NLP_subproblem(solve_data, config)
+
+        # add oa cuts
+        if fix_nlp_result.solver.termination_condition is tc.optimal:
+            self.handle_lazy_NLP_subproblem_optimal(
+                fix_nlp, solve_data, config, opt)
+        elif fix_nlp_result.solver.termination_condition is tc.infeasible:
+            self.handle_lazy_NLP_subproblem_infeasible(
+                fix_nlp, solve_data, config, opt)
+        else:
+            # TODO
```

Comment:
I guess there should be something here?

Reply:
Thanks for reminding me. I have added `self.handle_lazy_NLP_subproblem_other_termination` here.